components/membership/membership-http-rpc-impl/src/main/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImpl.kt

```
@@ -311,4 +331,53 @@ class CertificatesRestResourceImpl @Activate constructor(
             ).bytes
         }
     }
+
+    private fun validateX500Name(x500Name: String): X500Principal {
+        return try {
+            X500Principal(x500Name)
+        } catch (e: IllegalArgumentException) {
+            throw InvalidInputDataException(
+                "The X500 name of the certificate is invalid: ${e.message}.",
+                mapOf("x500Name" to x500Name)
+            )
+        }
+    }
+    private fun validateNodeSessionCertificateSubject(x500Name: String): MemberX500Name {
+        return try {
+            MemberX500Name.parse(x500Name)
+        } catch (e: IllegalArgumentException) {
+            throw InvalidInputDataException(
+                "The X500 name of the certificate is invalid: ${e.message}.",
```

Should we throw a different exception here, from the one on line 340? Something like `The X500 name of the certificate is not a valid Corda X500 name.`